projects/common/lib/components/button/button.component.ts

```diff
@@ -0,0 +1,58 @@
+import { forwardRef,Component, OnInit, ViewChild, EventEmitter, Input, Output, ElementRef } from '@angular/core';
+import { ControlContainer, ControlValueAccessor, NgForm, NG_VALUE_ACCESSOR } from '@angular/forms';
+
+/**
+ * Button Component is a button which can be used across the application to have same button style. 
+ * 
+ * 
+ * @example
+ *       	<common-button [label]="'Remove Spouse'" 
+ *            [styleClass]="buttonClass" 
+ *            (btnClick)='removeSpouse()'>
+ *        </common-button>
+ * @export
+ */
+@Component({
+  selector: 'common-button',
+  templateUrl: './button.component.html',
+  styleUrls: ['./button.component.scss'],
+  viewProviders: [
+    { provide: ControlContainer, useExisting: forwardRef(() => NgForm ) }
+  ],
+  providers: [
+    { provide: NG_VALUE_ACCESSOR, multi: true, useExisting: forwardRef(() => ButtonComponent )}
+  ]
+})
+
+export class ButtonComponent implements OnInit, ControlValueAccessor {
```

Comment:
You can remove ControlValueAccessor from ButtonComponent.

ControlValueAccessor is used for reactive forms to to make something a form control with a value.  This button doesn't have a value, it's just a button. It wouldn't attach to a form (NgForm) object, but rather it'd be used for something like "Add new" or whatever, which is app logic.

The clue here is that there's nothing to do in writeValue().

Reply:
Since the button will be clicked. Don't you think we might need _onTouched? I understand _onchanged and writeValue doesn't make any sense. Let me know your thoughts.

Reply:
Good question. I just did some Googling and we should be good to remove it regardless. The button will still be included in angular's regular change detection cycle. So, if we clicked this button, and the callback for it changed a form value (e.g. cleared the person's address), it would still properly update the UI to the user. 

What registerOnTouched() is for, is for telling if a given input is "touched" or "dirty" etc, so we can show validation errors. For example if an input is touched but empty we might say "this field is required.". We don't need to track that for a simple button.

Reply:
Cool, thanks!